app/controllers/admin/editions_controller.rb

```diff
@@ -74,6 +74,10 @@ def confirm_export
   end
 
   def show
+    if @edition.locked?
+      render :locked
+    end
```

Comment:
you probably want to return here as well to avoid executing the rest of the code. I would wonder if you want to have a separate action for this `show_locked` and redirect the show to this to keep them fully distinct.

Reply:
I've got this to call a `show_locked` method which renders the view instead of creating a separate route for it, does that sound ok?

Reply:
Hmm looking at that I think that ends up creating a method that you'd assume is an action.

If we're to go down the non-redirect route I think this would actually be more consistent with existing behaviours to have a before_action handle this conditional.

Reply:
I think I've got the redirect working properly now 👍